### DISCUSSION TITLE
Adding multicast feature

### DISCUSSION THREAD

**Discussion by AliReshadF** [Networking-for-Arduino/EthernetENC#66] [Q&A] [answered]
Hello!
I am working on a complex project with Esp32 that uses The EthernetENC library and I wanted to add the multicast feature to using mDNS benefits for accessing my board with hostname.local. So I searched the internet and reached these results:

EthernetENC has been made on uip stack and this stack has this documentation:

> uIP current supports broadcast packets as well as
> sending multicast packets. Joining multicast groups (IGMP) and
> receiving non-local multicast packets is not currently supported.

So for this reason we cannot join multicast groups in a local network. I didn't disappoint and continued to find other ways to have multicast.
Long story short I reached these ways:
1. Changing uip stack codes to handle IGMP messages (I think it needs low-level programming and a lot of effort I guess)
2. Switch to another library for example EthernetESP32 (I don't have any idea how much the two libraries are similar and if I do this, it will needs to change something in my project or not)

I should mention my project uses platform IO and has been not updated to framework version 3 yet.

In the end, I wanted to ask this question: 
Is it better to start changing uip stack codes to have multicast or wait for the Platform IO framework to update version 3 and use EthernetESP32?

**Comment by JAndrassy** [comment, marked as answer]
the uIP is almost 25 years old. maybe you can find a modification which supports multicast receive. 

the ESP32 networking uses LwIP which is actively maintained. the EthernetESP32 library integrates with the ESP32 Arduino v3 networking
